test/unit/modunit_pico_mld.c

```diff
@@ -118,16 +119,12 @@ END_TEST
 START_TEST(tc_pico_mld_report_expired)
 {
     struct mld_timer *t = PICO_ZALLOC(sizeof(struct mld_timer));
-    struct pico_ip6 zero = {{0}};
-    struct mcast_parameters p;
+    struct pico_ip6 zero = {0};
```

Comment:
my gcc version : gcc (Ubuntu 4.8.4-2ubuntu1~14.04.3) 4.8.4
In the thread you posted there seems to be some discussions going on about whether to backport it to 4.8 (and 4.7) or not - guess that didn't happen then.

Do I understand this correctly, the `{{0}}` version gives warnings on the most recent GCCs, while the `{0}` gives warnings on the older versions? If that's the case, and we can't find a nice and clean alternative, I think we have to follow the most recent GCC. 
But don't we have this sort of code in the stack itself? That does compile warning free on both older and newer GCC versions...